opm/grid/MinpvProcessor.hpp

```diff
@@ -159,28 +164,30 @@ namespace Opm
                             int c_above = ii + dims_[0] * (jj + dims_[1] * (kk - 1));
 
                             // Bypass inactive cells with thickness below tolerance and active cells with volume below minpv
-                            if (((actnum.empty() || !actnum[c_above]) && thickness[c_above] < z_tolerance) || ((actnum.empty() || actnum[c_above]) && pv[c_above] < minpvv[c_above]) ) {
+                            if (((actnum.empty() || !actnum[c_above]) && thickness[c_above] < z_tolerance) || ((actnum.empty() || actnum[c_above]) && pv[c_above] < minpvv[c_above]
```

Comment:
I think the logic is correct here, and can be merged. However, it was already hard to read, and became even harder now. I suggest making a few bool variables (active_above, thin_above, smallvolume_above for example) to improve readability and reduce the possibility of mistakes. I think the way things are written right now is a bit more complicated than necessary, it should not be necesssary to evaluate the minpvv condition when pinchNOGAP is true.

There is one minor thing that puzzles me: we seem to be inconsistent about the tests, using `<` and then later `>` rather than `>=` for the converse. That would automatically be addressed by using bool variables, and possibly cause a change in behaviour, if we were unlucky enough to depend on this (unlikely though).

Nitpick: line 190 indentation change?

Reply:
Yeah, that brings me straight to #472. Either the conditionals are wrong or the comment.
`(actnum.empty() || !actnum[c_above])` means to me "(all cells are active due to no ACTNUM) or cell above is inactive)".
I will do your suggestion as this makes the problems clearer.

Reply:
I understand #472 better now. I do not know the correct behaviour, but agree that it looks wrong. (On the other hand, I suspect that this code is never called with an empty actnum.)

Reply:
I did break down the logical statements.

Concerning not evaluating minpvv for pinch NOGAP. Currently, this reproduces the test and changing it would make the logic even more involved. If you case then please provide a test showing that this is wrong (I cannot really judge, here).
In that case i am happy to change.